tapdb/universe_stats.go

```diff
@@ -750,13 +752,18 @@ func (u *UniverseStats) QuerySyncStats(ctx context.Context,
 
 	// First, check the cache to see if we already have a cached result for
 	// this query.
-	syncSnapshots := u.syncStatsCache.fetchQuery(q)
+	var syncSnapshots []universe.AssetSyncSnapshot
+
+	u.syncStatsMtx.RLock()
+	syncSnapshots = u.syncStatsCache.fetchQuery(q)
```

Comment:
nit: can just use the `:=` syntax here, then you don't need to declare the `var` above.